src/zk.py

```diff
@@ -99,13 +100,8 @@ def __init__(self, config, plugins):
             self._path_prefix = prefix if prefix is not None else helpers.get_lockpath_prefix()
             self._lockpath = self._path_prefix + self.PRIMARY_LOCK_PATH
 
-            self._create_kazoo_client()
-            event = self._zk.start_async()
-            event.wait(self._timeout)
-            if not self._zk.connected:
+            if self._init_client():
```

Comment:
It looks like there is an error here and an error is thrown if the connection is successful.

https://github.com/bozaro/pgconsul/commit/8b837343582888c225c062c114e0b376e6ed5cfb